### PR TITLE
typescript(vx-text): Re-write package in TypeScript

### PR DESCRIPTION
**Pull request by williaster**
#### :rocket: Enhancements

This PR builds off of #488 which added TypeScript build support, and converts the `@vx/text` package to TypeScript.

#### Testing

- [x] CI
- [x] Functional (via `/gallery` and `/text` demo)
- [x] Generates `.d.ts` files

@hshoff @milesj @schillerk @techniq @kristw

### CODE REVIEW

File: packages/vx-text/package.json [hunk 1]
```diff
@@ -30,6 +31,7 @@
   },
   "homepage": "https://github.com/hshoff/vx#readme",
   "dependencies": {
+    "@types/react": "^16.9.2",
```
Comment (williaster) [inline packages/vx-text/package.json]: noting that we should make this less strict

File: packages/vx-text/src/Text.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      transform = transforms.join(' ');
+    }
+
+    return (
+      <svg
+        ref={innerRef}
+        x={dx}
+        y={dy}
+        fontSize={textProps.fontSize}
+        style={{ overflow: 'visible' }}
```
Comment (kristw) [inline packages/vx-text/src/Text.tsx]: make `style` constant?

File: packages/vx-text/src/util/getStringWidth.ts [hunk 3]
```diff
@@ -2,14 +2,14 @@ import memoize from 'lodash/memoize';
 
 const MEASUREMENT_ELEMENT_ID = '__react_svg_text_measurement_id';
 
-function getStringWidth(str, style) {
+function getStringWidth(str: string, style?: Object) {
```
Comment (milesj) [inline packages/vx-text/src/util/getStringWidth.ts]: Use `object`.

Comment (williaster) [reply packages/vx-text/src/util/getStringWidth.ts]: whoops not intentional

**Pull request merged**